### PR TITLE
Adding subscriber name to CREATED_BY column value of AM_SUBSCRIPTION table

### PR DESCRIPTION
**Pull request by Sarangan0219**
Fixes https://github.com/wso2/product-apim/issues/7933

### CODE REVIEW

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/dao/ApiMgtDAO.java [hunk 1]
```diff
@@ -807,7 +807,7 @@ public Subscriber getSubscriber(int subscriberId) throws APIManagementException
         return null;
     }
 
-    public int addSubscription(ApiTypeWrapper apiTypeWrapper, int applicationId, String status)
+    public int addSubscription(ApiTypeWrapper apiTypeWrapper, int applicationId, String status, String subscriber)
```
Comment (uvindra) [inline components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/dao/ApiMgtDAO.java:810]: subscriber parameter defined but not used. I think you need to insert the value into DB when adding the subscription to fix the bug.

Comment (Sarangan0219) [reply components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/dao/ApiMgtDAO.java:810]: https://github.com/wso2/carbon-apimgt/pull/8531/commits/32efd32edd3c2ca791281eea000f281341d39bd6

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/dao/ApiMgtDAO.java [hunk 2]
```diff
@@ -5788,7 +5788,8 @@ public void makeKeysForwardCompatible(ApiTypeWrapper apiTypeWrapper, String oldV
                             subscriptionStatus = APIConstants.SubscriptionStatus.ON_HOLD;
                         }
                         apiTypeWrapper.setTier(info.tierId);
-                        int subscriptionId = addSubscription(apiTypeWrapper, info.applicationId, subscriptionStatus);
+                        int subscriptionId = addSubscription(apiTypeWrapper, info.applicationId, subscriptionStatus,
+                                apiIdentifier.getProviderName());
```
Comment (praminda) [inline components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/dao/ApiMgtDAO.java:5792]: subscriber and provider are two different roles. `apiIdentifier.getProviderName()` is this the correct value to be set here?

Comment (Sarangan0219) [reply components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/dao/ApiMgtDAO.java:5792]: In this context API provider is the subscriber.

**Pull request merged**